docs/build/guides/transactions/invoke-contract-tx-sdk.mdx

```diff
@@ -18,117 +18,111 @@ Please go to [the project homepage](https://github.com/stellar/js-stellar-sdk) o
 
 :::
 
-```javascript
-(async () => {
-  const {
-    Keypair,
-    Contract,
-    SorobanRpc,
-    TransactionBuilder,
-    Networks,
-    BASE_FEE,
-    nativeToScVal,
-    Address,
-  } = require("@stellar/stellar-sdk");
+First, upload the bytes of [the example contract](https://developers.stellar.org/docs/build/smart-contracts/example-contracts/auth) onto the blockchain using [Stellar CLI](../../../tools/developer-tools/cli/stellar-cli). This is called "install" because, from the perspective of the blockchain itself, this contract has been installed.
+
+```bash
+stellar contract build
+stellar contract install --wasm target/wasm32-unknown-unknown/release/test_auth_contract.wasm --network testnet
+```
+
+This will return a hash; save that, you'll use it soon. For this example, we will use `bc7d436bab44815c03956b344dc814fac3ef60e9aca34c3a0dfe358fcef7527f`.
+
+No contract has yet been deployed with this hash. In Soroban, you can have many Smart Contracts which all reference the same Wasm hash, defining their behavior. We'll do that from the following JavaScript code itself.
+
+```ts
+import { Keypair } from "@stellar/stellar-sdk"
+import { Client, basicNodeSigner } from "@stellar/stellar-sdk/contract"
 
+// As mentioned, we are using Testnet for this example
+const rpcUrl = "https://soroban-testnet.stellar.org"
+const networkPassphrase = "Test SDF Network ; September 2015"
+const wasmHash = "bc7d436bab44815c03956b344dc814fac3ef60e9aca34c3a0dfe358fcef7527f"
+
+/**
+ * Generate a random keypair and fund it on Testnet
+ */
+async function generateFundedKeypair() {
+  const keypair = Keypair.random();
+  await fetch(`https://friendbot.stellar.org/?addr=${
```

Comment:
I wish this was a method on Keypair or an account type?

Reply:
Looks like there's a [server.requestAirdrop]() function that does exactly that. Terrible name, though!

Reply:
Tried out using `requestAirdrop` here

- https://github.com/stellar/js-stellar-sdk/pull/1125